compiler/frontend/beam/src/main/java/org/apache/nemo/compiler/frontend/beam/PipelineTranslationContext.java

```diff
@@ -55,6 +56,7 @@
   private final Map<PValue, TupleTag<?>> pValueToTag;
   private final Stack<LoopVertex> loopVertexStack;
   private final Pipeline pipeline;
+  private AppliedPTransform<?, ?, ?> currentTransform;
```

Comment:
Can you perhaps do without this? Something like this has not been needed, because the current transform that's being visited is explicitly provided through the Beam PipelineVisitor interface.

Reply:
If you look at the [PR](https://github.com/apache/beam/pull/7635/files) it's been implemented this way, and that's why I've implemented this in the way described here, but I'll try to follow your comments as I think that could be a better approach.